### ISSUE TITLE
no1_D6 kernel ?

### ISSUE THREAD

**Issue by ranshalit** [corvusmod/device_no1_D6#1]
Hello corvusmod,

Is there D6 kernel source available ?

Best Regards,
Ranchu

**Comment by corvusmod** [comment]
As much as I know there is not public kernel sources. I'm using prebuilt
one.

**Comment by ranshalit** [comment]
Ok, Thank you very much !

One more thing please:
device_no1_D6 - Is it partitial so libraries ?
vendor_no1_D6 - file system for D6 (I think it is also public as part of ROM)

Many thanks,
Ranchu

**Comment by corvusmod** [comment]
This device is not working by now. I'm trying to make it work, but only get
adb. :(



---------------------------------------------------------------------------------------------
Que todos esos ceros y unos vayan directamente a tu mierdecilla de teléfono
inteligente y que cualquier inutil sea capaz de suscribirse en 12 segundos *no
es «magia», es sudor y talento.*

**Comment by ranshalit** [comment]
Thank you very much,
I also try to contact NO1 company to see if they can public kernel. 
If they will, I'll update here.

**Comment by corvusmod** [comment]
Thanks a lot


---------------------------------------------------------------------------------------------
Que todos esos ceros y unos vayan directamente a tu mierdecilla de teléfono
inteligente y que cualquier inutil sea capaz de suscribirse en 12 segundos *no
es «magia», es sudor y talento.*

**Comment by ranshalit** [comment]
Hi Again,

Maybe you'll have an idea for me...
I am trying to make the screen turn off at startup. I had tried to force device shut down in many ways, but all failed. I also try to disable touch screen, but didn't succeed with that. seems that the only alternative to make both work would be to use the kernel source code. Unless I find another way to do that. Do you think of any possibility to achieve this ? Thx .

**Comment by corvusmod** [comment]
But, what are you trying to do? Why do you want to disable touch screen and
screen turn off?

To disable the touch screen, just root your device (search in xda) and
delete some lib with "tp" in the filename. But I don't know if the device
will boot.

---------------------------------------------------------------------------------------------
Que todos esos ceros y unos vayan directamente a tu mierdecilla de teléfono
inteligente y que cualquier inutil sea capaz de suscribirse en 12 segundos *no
es «magia», es sudor y talento.*

**Comment by ranshalit** [comment]
Hi,

Thanks for the ideas, removing the libraries is good idea, I've tried that idea. As you said, some libraries removal end up with non bootable device.
Therefore, my other alternative is to make the changes in:
1. Hardware libraries - I think they are not available for D6  anywhere, Right ? As a matter of fact I thought that this git has the hardware libs for D6, but I think it's not, Right ?

OR

2. Anrdoid framework - I think that using google's framework 5.1.0 should be good enough for D6, or do you think I should use any other git ?

Thx for your time!

**Comment by corvusmod** [comment]
You are correct, there is no kernel or harware sources for modify :(

And in android framework, from what I know this is the only public sorces.
:(

So if we don't get official sources I'm afraid that you will not be able to
do anything :(

---------------------------------------------------------------------------------------------
Que todos esos ceros y unos vayan directamente a tu mierdecilla de teléfono
inteligente y que cualquier inutil sea capaz de suscribirse en 12 segundos *no
es «magia», es sudor y talento.*

**Comment by ranshalit** [comment]
Hi,

Do you mean that NO.1  might also made modifications in  android 5.1 framework for D6 ?
The spec say that it use android 5.1 ...
Isn't it the android-5.1 from google's repo ? 

Thx

**Comment by corvusmod** [comment]
All mediatek (and not mediatek) devices has changes to android repos to
make android working in their hardware. And these changes are not public :(

---------------------------------------------------------------------------------------------
Que todos esos ceros y unos vayan directamente a tu mierdecilla de teléfono
inteligente y que cualquier inutil sea capaz de suscribirse en 12 segundos *no
es «magia», es sudor y talento.*

**Comment by ranshalit** [comment]
Hi,

Please take a look at this:
[https://android.googlesource.com/kernel/mediatek/+/android-5.1.1_r0.16](https://android.googlesource.com/kernel/mediatek/+/android-5.1.1_r0.16)

Isn't it the mediatek kernel 5.1 ? NO1 D6 is also mediatek (65xx) , android 5.1.
So isn't it just the same kernel in D6 ?

Thanks,
Ranchu

**Comment by corvusmod** [comment]
Yes, this is a generic mediatek kernel, but for each specific device the manufacturer (No1 in the case of D6) adds some custom parts to support the hardware, and without this modifications the kernel will not boot.

**Comment by ranshalit** [comment]
I think the manufacture must give the sources if being asked to do that, because of licenses issue (GPL).
see here:
https://www.reddit.com/r/Android/comments/4ww4vu/psa_you_can_get_the_kernel_source_for_every/

**Comment by corvusmod** [comment]
Yes, they should but usually they doesnt do 😑
